homeassistant/components/sensor/__init__.py

```diff
@@ -246,6 +247,22 @@ def device_class(self) -> SensorDeviceClass | None:
             return self.entity_description.device_class
         return None
 
+    @final
+    @property
+    def numeric_state_expected(self) -> bool:
+        """Return true if the sensor is expected to be numeric."""
+        if (
+            self.state_class is not None
+            or self.native_unit_of_measurement is not None
+            or self.native_precision is not None
+        ):
+            return True
+        device_class: SensorDeviceClass | None = None
+        with suppress(ValueError):
+            # Custom device classes are not considered numeric
+            device_class = SensorDeviceClass(str(self.device_class))
+        return bool(device_class and device_class not in NON_NUMERIC_DEVICE_CLASSES)
```

Comment:
The use of `bool()` really isn't needed here

Reply:
It is, when `device_class` is `None` it will return `None`

Reply:
It isn't the reason it is now needed, is because the comparison on the first element is missing.